lib/fluent/plugin/in_tail.rb

```diff
@@ -969,11 +980,12 @@ def handle_notify
                       read_more = true
                       break
                     end
-                    if limit_bytes_per_second_reached?(start_reading, number_bytes_read)
+                    if limit_bytes_per_second_reached?(start_reading, @number_bytes_read)
                       # Just get out from tailing loop.
                       read_more = false
                       break
                     end
+                    @last_read_time = Fluent::Clock.now
```

Comment:
Probably storing `start_reading` and reset also it in `refresh_bytes_read_counter` is more appropriate.
By this, `limit_bytes_per_second_reached?` doesn't need arguments any more.

Reply:
That's reasonable.
I decided to make `start_reading` as instance variable to store starting to read time.